tests/core/test_evaluation_parameters.py

```diff
@@ -306,3 +307,46 @@ def test_evaluation_parameters_for_between_expectations_parse_correctly(
             "result": {"observed_value": 3},
         }
     )
+
+
+def test_build_evaluation_parameters():
+    evaluation_parameters = {
+        "my_min": 1,
+        "my_max": 5,
+    }
+    expectation_args = {
+        "min_value": {"$PARAMETER": "my_min", "$PARAMETER.upstream_row_count": 10},
+        "max_value": {"$PARAMETER": "my_max", "$PARAMETER.upstream_row_count": 50},
+    }
+    res = build_evaluation_parameters(
+        expectation_args, evaluation_parameters, interactive_evaluation=True
+    )
+    expected = {"max_value": 5, "min_value": 1}
+    assert res[0] == expected
+    assert res[1] == expected
+
+
+def test_build_evaluation_parameters_with_nested_PARAMETERS():
+    evaluation_parameters = {
+        "my_min": 1,
+        "my_max": 5,
+    }
+    expectation_args = {
+        "min_value": {
+            "foo": {
+                "$PARAMETER": "my_min",
+                "$PARAMETER.upstream_row_count": 10,
+            }
+        },
+        "max_value": {
+            "bar": {
+                "$PARAMETER": "my_max",
+                "$PARAMETER.upstream_row_count": 50,
+            }
+        },
+    }
+    res = build_evaluation_parameters(
+        expectation_args, evaluation_parameters, interactive_evaluation=True
+    )
+    assert res[0] == {"max_value": {"bar": 5}, "min_value": {"foo": 1}}
+    assert res[1] == {"bar": 5, "foo": 1}
```

Comment:
Didn't have the best test case here. Any thoughts on how to possible improve this?

Reply:
Could we add something more like an integration test? Something like two validators, the first validates a suite, and the second gets some nested evaluation params out of the first?